src/parse/spec.js

```diff
@@ -3,11 +3,58 @@ var dl = require('datalib'),
     Model = require('../core/Model'),
     View = require('../core/View');
 
-function parseSpec(spec, callback) {
-  var vf = arguments[arguments.length-1],
+/**
+ * Backward compatibility wrapper that accepts callback without error handler
+ * @param spec (object)
+ * @param callback (model)
+ * @param config (optional object)
+ * @param viewFactory (optional callback)
+ * @returns {*}
+ */
+function parseSpec(spec, callback /* [, config] [, viewFactory] */) {
+  var cb = callback,
+      vf = arguments[arguments.length-1],
       viewFactory = arguments.length > 2 && dl.isFunction(vf) ? vf : View.factory,
-      config = arguments[2] !== viewFactory ? arguments[2] : {},
-      model = new Model(config);
+      config = arguments[2] !== viewFactory ? arguments[2] : {};
+
+  return module.exports.parse(spec, config, viewFactory, function(err, model) {
+    // For backward compatibility, the error is thrown even though it might never be caught
+    if (err) throw err;
+    cb(model);
+  });
+}
+
+module.exports = parseSpec;
+
+/**
+ * Parse graph specification
+ * @param spec (object)
+ * @param config (optional object)
+ * @param viewFactory (optional function)
+ * @param callback (error, model)
+ */
+parseSpec.parse = function (spec, /* [config,] [viewFactory,] */ callback) {
+  // do not assign any values to callback, as it will change arguments
+  var cb = arguments[arguments.length - 1],
+      model, argInd = 2,
+      viewFactory = View.factory;
+
+  function done(err, value) {
+    if (cb) {
+      cb(err, value);
+      cb = undefined;
+    }
+  }
```

Comment:
Second, why is `done` needed? Why not call `cb` directly where needed? (And perhaps throw an error if no cb was specified?) Moreover, why set the cb as `undefined` at the end? Thanks!